app/src/main/java/de/danoeh/antennapod/fragment/AllEpisodesFragment.java

```diff
@@ -61,36 +56,33 @@ private void inflateSortMenu() {
     @NonNull
     @Override
     protected List<FeedItem> loadData() {
-        return DBReader.getEpisodes(0, page * EPISODES_PER_PAGE, getFilter(), getSortOrder());
+        return DBReader.getEpisodes(0, page * EPISODES_PER_PAGE, getFilter(),
+                UserPreferences.getAllEpisodeSortOrder());
     }
 
     @NonNull
     @Override
     protected List<FeedItem> loadMoreData(int page) {
-        return DBReader.getEpisodes((page - 1) * EPISODES_PER_PAGE, EPISODES_PER_PAGE, getFilter(), getSortOrder());
+        return DBReader.getEpisodes((page - 1) * EPISODES_PER_PAGE, EPISODES_PER_PAGE, getFilter(),
+                UserPreferences.getAllEpisodeSortOrder());
     }
 
     @Override
     protected int loadTotalItemCount() {
         return DBReader.getTotalEpisodeCount(getFilter());
     }
 
+
     @Override
     protected FeedItemFilter getFilter() {
-        SharedPreferences prefs = getActivity().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
-        return new FeedItemFilter(prefs.getString(PREF_FILTER, ""));
+        return new FeedItemFilter(UserPreferences.getPrefFilterAllEpisodes());
     }
 
     @Override
     protected String getFragmentTag() {
         return TAG;
     }
 
-    @Override
-    protected String getPrefName() {
-        return PREF_NAME;
```

Comment:
I think this should stay, that's where the parent class stores the scroll position

Reply:
Thank you, I learned something new